sdk/core/corehttp/corehttp/runtime/policies/_authentication.py

```diff
@@ -32,15 +32,22 @@ class _BearerTokenCredentialPolicyBase:
     :param credential: The credential.
     :type credential: ~corehttp.credentials.TokenCredential
     :param str scopes: Lets you specify the type of access needed.
+    :keyword list[dict[str, str]] auth_flows: A list of authentication flows to use for the credential.
     """
 
+    # pylint: disable=unused-argument
     def __init__(
-        self, credential: "TokenCredential", *scopes: str, **kwargs: Any  # pylint: disable=unused-argument
+        self,
+        credential: "TokenCredential",
+        *scopes: str,
+        auth_flows: Optional[list[dict[str, str]]] = None,
```

Comment:
@xiangyan99 the flows also contains scopes:
![image](https://github.com/user-attachments/assets/dae6938c-84b2-45d2-8faf-da3446d467e4)
And we can't expect whether there will be other type in the future so is it better for typing to be `Optional[list[dict[str, Any]]]`?

Reply:
Yes. We need all the information to be passed into the ctor. I will update the typing. Is `scopes` always a list?

Reply:
According to current definition in @typespec/http, it is always list : https://github.com/microsoft/typespec/blob/bf528cdd57a38bb25e61065c79509a5d5831e631/packages/http/src/types.ts#L132-L168. But I am not sure whether there will be other types added in the flow.

Reply:
@xiangyan99 I find that `OAuth2Scope` is an [object](https://github.com/microsoft/typespec/blob/9faa31eb767c9ef00a9ef28b4bbeab955b7d1a94/packages/http/src/types.ts#L170-L173) instead of string, but tsp compiler will report error. I have created an issue for typespec team https://github.com/microsoft/typespec/issues/6692. Considering typespec is not stable, I still suggestion we type it with `Optional[list[dict[str, Any]]]`.